src/NET8MinimalAPISnapSnart/ApiBootstrap/Function.cs

```diff
@@ -0,0 +1,180 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text.Json;using System.Threading.Tasks;
+using Amazon.CloudWatchLogs;
+using Amazon.CloudWatchLogs.Model;
+using ApiBootstrap;
+using GetProducts;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Shared;
+using Shared.DataAccess;
+using Shared.Models;
+
+var builder = WebApplication.CreateSlimBuilder(args);
+builder.Services.AddServices(builder.Configuration);
+builder.Services.AddSingleton<Handlers>();
+builder.Logging.ClearProviders();
+builder.Logging.AddJsonConsole(options =>
+{
+    options.IncludeScopes = true;
+    options.UseUtcTimestamp = true;
+    options.TimestampFormat = "hh:mm:ss ";
+});
+
+var app = builder.Build();
+
+var dataAccess = app.Services.GetRequiredService<ProductsDAO>();
+var handlers = app.Services.GetRequiredService<Handlers>();
+
+Amazon.Lambda.Core.SnapshotRestore.RegisterBeforeSnapshot(async () => await BeforeCheckpoint(app.Logger, handlers));
+Amazon.Lambda.Core.SnapshotRestore.RegisterBeforeSnapshot(AfterRestore);
+
+var cloudWatchClient = new AmazonCloudWatchLogsClient();
+
+app.MapGet("/", async () => await handlers.ListProducts());
+
+app.MapDelete("/{id}", async (HttpContext context) =>
+{
+    try
+    {
+        var id = context.Request.RouteValues["id"].ToString();
```

Comment:
Is there a reason that Delete and Put don't have their own handlers? We can always add them later if we want though. Also, then we could more easily warm them maybe.

Reply:
Only because I didn't want to actually mutate data in the database on statup. I suppose we could create, update and delete a record as part of the startup.